tests/tests/swfs/avm2/edittext_newline_character/Test.as

```diff
@@ -0,0 +1,40 @@
+package {
+    import flash.display.Sprite;
+    import flash.events.Event;
+    import flash.events.KeyboardEvent;
+    import flash.events.TextEvent;
+    import flash.text.TextField;
+    import flash.text.TextFieldType;
+
+    public class Test extends Sprite {
+        public function Test() {
+            var textField: TextField = new TextField();
+            textField.type = TextFieldType.INPUT;
+            textField.border = true;
+            textField.multiline = true;
+            textField.height = 200;
+            textField.addEventListener(Event.CHANGE, onChange);
+            textField.addEventListener(KeyboardEvent.KEY_DOWN, onKeyDown);
+            textField.addEventListener(TextEvent.TEXT_INPUT, onTextInput);
+            addChild(textField);
+        }
+
+        public function onChange(evt: Event): void {
+            trace("onChange");
+
+            var text: String = evt.target.text;
+            for (var i: int = 0; i < text.length; i++) {
+                trace(" U+" + text.charCodeAt(i).toString(16));
+            }
+        }
+
+        public function onKeyDown(evt: KeyboardEvent): void {
+            // TODO: We must not be firing this event yet.
```

Comment:
Why does this comment mean? Why can't you just fire the event?

Reply:
Oh, sorry, I misunderstood the comment. I thought we cannot fire the event here somehow.

The key down/up events are separate from text input events, you need to add them to `input.json`.

Reply:
Do we not have a way of producing the TEXT_INPUT and KEY_DOWN event with just one line in `input.json`? It seems a bit pointless to test if both results are basically hardcoded. (We also don't seem to support specific a character for test KeyDown events, which is another problem)

Reply:
> Do we not have a way of producing the TEXT_INPUT and KEY_DOWN event with just one line in `input.json`?

Nope

> It seems a bit pointless to test if both results are basically hardcoded.

Text input is inherently different from key up/down, because it goes through the OS (keyboard layout, mapping, etc.). We cannot infer one from the other, it's done outside of Ruffle.

> We also don't seem to support specifying a character for test KeyDown events, which is another problem

I agree, I remember adding key char as an optional field in the automated event. Maybe I'll find the commit

Reply:
Oh, I remember now. I worked on redoing key events in Ruffle to properly support key mapping. I believe we should do it first, because key code can be inferred from key char (contrary to text input).

Reply:
I think we have to add KeyDown/KeyUp events to `input.json` and we cannot combine them with `TextInput` due to how input works on Windows: https://github.com/ruffle-rs/ruffle/pull/19874